tests/integ/scala/test_dataframe_join_suite.py

```diff
@@ -420,39 +441,40 @@ def test_semi_join_with_columns_from_LHS(session):
     assert sorted(res, key=lambda x: x[0]) == [Row(1), Row(2)]
 
 
-def test_using_joins(session):
+@pytest.mark.localtest
+@pytest.mark.parametrize("join_type", ["inner", "leftouter", "rightouter", "fullouter"])
+def test_using_joins(session, join_type):
     lhs = session.create_dataframe([[1, -1, "one"], [2, -2, "two"]]).to_df(
         ["intcol", "negcol", "lhscol"]
     )
     rhs = session.create_dataframe([[1, -10, "one"], [2, -20, "two"]]).to_df(
         ["intcol", "negcol", "rhscol"]
     )
 
-    for join_type in ["inner", "leftouter", "rightouter", "full_outer"]:
-        res = lhs.join(rhs, ["intcol"], join_type).select("*").collect()
-        assert res == [
-            Row(1, -1, "one", -10, "one"),
-            Row(2, -2, "two", -20, "two"),
-        ]
+    res = lhs.join(rhs, ["intcol"], join_type).select("*").collect()
+    assert res == [
+        Row(1, -1, "one", -10, "one"),
+        Row(2, -2, "two", -20, "two"),
+    ]
 
-        res = lhs.join(rhs, ["intcol"], join_type).collect()
-        assert res == [
-            Row(1, -1, "one", -10, "one"),
-            Row(2, -2, "two", -20, "two"),
-        ]
+    res = lhs.join(rhs, ["intcol"], join_type).collect()
+    assert res == [
+        Row(1, -1, "one", -10, "one"),
+        Row(2, -2, "two", -20, "two"),
+    ]
 
-        with pytest.raises(SnowparkSQLAmbiguousJoinException) as ex_info:
-            lhs.join(rhs, ["intcol"], join_type).select("negcol").collect()
-        assert "reference to the column 'NEGCOL' is ambiguous" in ex_info.value.message
-
-        res = lhs.join(rhs, ["intcol"], join_type).select("intcol").collect()
-        assert res == [Row(1), Row(2)]
-        res = (
-            lhs.join(rhs, ["intcol"], join_type)
-            .select(lhs["negcol"], rhs["negcol"])
-            .collect()
-        )
-        assert sorted(res, key=lambda x: -x[0]) == [Row(-1, -10), Row(-2, -20)]
+    # with pytest.raises(SnowparkSQLAmbiguousJoinException) as ex_info:
+    #    lhs.join(rhs, ["intcol"], join_type).select("negcol").collect()
+    # assert "reference to the column 'NEGCOL' is ambiguous" in ex_info.value.message
```

Comment:
@sfc-gh-stan , should we use `if local_testing_mode` here to get the error behavior check work for live session?

Reply:
uncommented